### ISSUE TITLE
UInt and UByte return values cannot be mocked in suspending functions

### ISSUE THREAD

**Issue by david-livefront** [mockk/mockk#1259]
### Prerequisites

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### Expected Behavior

When I create a mock for a Kotlin class, I should be able to mock a `UInt` or `UByte` response on a suspending function.

### Current Behavior

The mock will return a Java Integer or Java Byte instead of the appropriate Kotlin unsigned type, this causes type mismatches and can lead to a `ClassCastException`.

#### Context

MockK version: 1.13.11
OS: MacOS Sonoma 14.4.1
Kotlin version: 1.9.24
JDK version: openjdk-17
JUnit version: 5.10.2
Type of test: unit test

#### Minimal reproducible code (the gist of this issue)

Note that removing the `suspend` keyword from the `bar()` function causes this test to pass.

```kotlin
    interface Foo {
        suspend fun bar(): UByte
    }

    @Test
    fun `mock Ubyte`() = runBlocking {
        val result: UByte = 5u
        val mock = mockk<Foo> {
            coEvery { bar() } returns result
        }

       assertEquals(result, mock.bar())
    }
```

**Comment by VasilisDrettas-tomtom** [comment]
I think this is fixed with the latest version of mockk 1.13.12. Could you try it again ?

**Comment by david-livefront** [comment]
Confirmed that it works as expected. Thanks!
